File: app/src/main/java/com/example/myaac/model/HomeButtonConfig.kt
package com.example.myaac.model

data class HomeButtonConfig(
    val id: String,
    val label: String,
    val iconPath: String? = null,
    val action: ButtonAction,
    val backgroundColor: Long = 0xFFE3F2FD,
    val order: Int = 0,
    val isVisible: Boolean = true
)

File: web/src/models/HomeButtonConfig.ts
import type { ButtonAction } from './ButtonAction';

/** Mirrors com.example.myaac.model.HomeButtonConfig. */
export interface HomeButtonConfig {
  id: string;
  label: string;
  iconPath: string | null;
  action: ButtonAction;
  /** Packed ARGB. Kotlin default is 0xFFE3F2FD (a light blue). */
  backgroundColor: number;
  order: number;
  isVisible: boolean;
}

export const DEFAULT_HOME_BUTTON_BACKGROUND_COLOR = 0xffe3f2fd;
